lib/Sema/TypeChecker.cpp

```diff
@@ -1003,6 +1003,37 @@ TypeChecker::getDeclTypeCheckingSemantics(ValueDecl *decl) {
   return DeclTypeCheckingSemantics::Normal;
 }
 
+ArrayRef<ValueDecl*> swift::
+canDeclProvideDefaultImplementationFor(ValueDecl* VD,
```

Comment:
If these kinds of utilities are only used by SourceKit, is it OK if you put them somewhere in IDE/ instead, to avoid cluttering Sema with one-off code that's not really part of the type checking itself?